garrysmod/gamemodes/base/entities/entities/base_trigger.lua

```diff
@@ -0,0 +1,95 @@
+
+ENT.Base = "base_entity"
+ENT.Type = "trigger"
+
+--[[---------------------------------------------------------
+	Name: Initialize
+-----------------------------------------------------------]]
+function ENT:Initialize()
+	self:SetTrigger(true)
+	self.Entities = {}
+end
+
+--[[---------------------------------------------------------
+	Name: IsTouchedBy
+-----------------------------------------------------------]]
+function ENT:IsTouchedBy(ent)
+	return self.Entities[ent]
```

Comment:
There's not really a problem with returning the entity, but if you cared I would prefer `return self.Entities[ent] != nil`